dracut/repartition/endless-repartition.sh

```diff
@@ -80,9 +80,18 @@ if [ -z "${root_disk}" ]; then
   exit 0
 fi
 
+pt_label=$(blkid -o value -s PTTYPE $root_disk)
+
 # Check for our magic "this is Endless" marker
-marker=$(sfdisk --force --part-attrs $root_disk $partno)
-if [ "$marker" != "GUID:55" ]; then
+if [ "$pt_label" = "dos" ]; then
+  marker=$(sfdisk --force --part-type $root_disk 4)
```

Comment:
Uhm, `4` is wrong but also `$partno` is not correct either. `$partno` is the partition number for the rootfs where is the marker for the GPT case. For the DOS case we have a new additional partition so it should be `$partno+1` if I'm not mistaken.